normative-conventions.md

```diff
@@ -0,0 +1,23 @@
+# Normative conventions
+
+This document describes some of the conventions that we try to follow when designing new features. Sometimes old features don't follow these; that's ok, new features generally should anyway.
+
+None of these rules are inviolable, but you should have a good reason for any particular feature to deviate.
+
+This list is very far from being complete.
+
+## Number-taking inputs should reject `NaN`
+
+In any place which expects a number, receiving `NaN` - or anything which coerces to `NaN`, if coercion is performed - should throw a `RangeError`.
```

Comment:
```suggestion
In any built-in function that expects to receive a Number, receiving `NaN` or anything which results in `NaN` after coercion is performed  (such as by passing through `ToNumeric`, `ToPrimitive`, `ToNumber`, etc.) should cause a `RangeError` to be thrown.
```

Reply:
I want this apply to options bags too, not just to bare arguments, but I'll keep the rest of it.

I probably need to update it to make it explicit that we check after _type_ coercion, not after e.g. `ToLength` coercion.

Reply:
I would read this as applying to options bags, but it doesn't hurt to be more explicit.

Reply:
Updated.